packages/api/test/e2e/staking.e2e.ts

```diff
@@ -0,0 +1,302 @@
+      const ledger = ledgerOpt.unwrapOr(null);
+      if (ledger?.active.toNumber() === (previousLedger.active.toNumber() + additionalBond)) {
+        done();
+      }
+    });
+
+    await api.tx.staking.bondExtra(additionalBond).signAndSend(stash);
+  });
+
+  test('unbond schedules some funds to unlock', async done => {
+    const unbondAmount = 500;
+    const previousLedger = ((await api.query.staking.ledger(controller.address)) as Option<StakingLedger>).unwrap();
+
+    // Subscribe to ledger value changes
+    await api.query.staking.ledger(controller.address, (ledgerOpt: Option<StakingLedger>) => {
+      const ledger = ledgerOpt.unwrapOr(null);
+      if (ledger?.active.toNumber() === (previousLedger.active.toNumber() - unbondAmount)) {
+        done();
+      }
+    });
+
+    await api.tx.staking.unbond(unbondAmount).signAndSend(controller);
+  });
+
+  /// Rebond a portion of the stash scheduled to be unlocked.
+  test('rebond locks funds again', async done => {
+    const rebondAmount = 300;
+    const previousLedger = ((await api.query.staking.ledger(controller.address)) as Option<StakingLedger>).unwrap();
+
+    // Subscribe to ledger value changes
+    await api.query.staking.ledger(controller.address, (ledgerOpt: Option<StakingLedger>) => {
+      const ledger = ledgerOpt.unwrapOr(null);
+      if (ledger?.active.toNumber() === (previousLedger.active.toNumber() + rebondAmount)) {
+        done();
+      }
+    });
+
+    await api.tx.staking.rebond(rebondAmount).signAndSend(controller);
+  });
+
+  test('withdraw unbonded', async done => {
+    await api.tx.staking.withdrawUnbonded().signAndSend(controller, ({ status, events }) => {
+      if (status.isInBlock) {
+        expect(
+          events.find(wrapper => wrapper.event.method === 'ExtrinsicSuccess')
+        ).toBeDefined();
+        done();
+      }
+    });
+  });
+
+  test('validate adds stash as a validator candidate', async done => {
+    // parts per billion
+    // 100,000,000 / 1,000,000,000 == 0.1%
+    const commission = 1_000_000_000;
+
+    const checkCommission = async ({ status }) => {
+      if (status.isInBlock) {
+        const prefs = ((await api.query.staking.validators(stash.address)) as ValidatorPrefs);
+        expect(prefs.commission.toNumber()).toEqual(commission);
+        done();
+      };
+    };
+
+    await api.tx.staking.validate({ commission }).signAndSend(controller, checkCommission);
+  });
+
+  test('chill removes stash from validator candidacy', async done => {
+
+    const checkCommission = async ({ status }) => {
+      if (status.isInBlock) {
+        const prefs = ((await api.query.staking.validators(stash.address)) as ValidatorPrefs);
+        expect(prefs.commission.toNumber()).toEqual(0);
+        done();
+      };
+    };
+
+    await api.tx.staking.chill().signAndSend(controller, checkCommission);
+  });
+
+  test('setPayee changes reward destination', async done => {
```

Comment:
Is controller account by default a payee account, I suppose there is a `getPayee` method to grab it, right?

Reply:
> Is controller account by default a payee account,
no, the payee account is originally set on a stash when bond() is called (it could be set to the stash itself or the controller)

> I suppose there is a getPayee method to grab it, right?
`api.query.staking.payee(<stash>)` is the getter method